subsys/net/ip/l2/ethernet/ethernet.c

```diff
@@ -757,18 +759,45 @@ int net_eth_vlan_disable(struct net_if *iface, u16_t tag)
 NET_L2_INIT(ETHERNET_L2, ethernet_recv, ethernet_send, ethernet_reserve,
 	    ethernet_enable);
 
+#if defined(CONFIG_NET_L2_ETHERNET_MGMT)
+static void carrier_handler(struct net_mgmt_event_callback *cb,
+			    u32_t mgmt_event, struct net_if *iface)
+{
+	if (mgmt_event == NET_EVENT_ETHERNET_CARRIER_ON) {
```

Comment:
so move this in a ethernet_carrier_on() function  (also call the net event for ethernet carrier on relevantly)